source/core/parameters_lagrangian.cc

```diff
@@ -175,70 +172,71 @@ namespace Parameters
                             friction_coefficient_particle,
                             rolling_friction_coefficient_particle,
                             surface_energy_particle);
-      {
-        g[0] = prm.get_double("gx");
-        g[1] = prm.get_double("gy");
-        g[2] = prm.get_double("gz");
 
-        particle_type_number = prm.get_integer("number of particle types");
+      g[0] = prm.get_double("gx");
+      g[1] = prm.get_double("gy");
+      g[2] = prm.get_double("gz");
+
+      particle_type_number = prm.get_integer("number of particle types");
 
-        if (particle_type_number >= 1)
+      if (particle_type_number >= 1)
+        {
+          prm.enter_subsection("particle type 0");
```

Comment:
Instead of having a ton of ifs like that, why not just make a single for loop?

Reply:
I will try to do something similar to the parsing of the solid object.

Reply:
Done